pages/admin/lessons/[lessonId]/[pageName]/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { gql, useQuery } from '@apollo/client'
+import { GetAppProps, withGetApp } from '../../../../../graphql'
+import { get, sortBy, toNumber, toUpper } from 'lodash'
+import React, { useMemo, useState } from 'react'
+import AdminLessonNav from '../../../../../components/admin/lessons/AdminLessonSideNavLayout'
+import AdminLessonSideNav from '../../../../../components/admin/lessons/AdminLessonSideNav'
+import AdminModuleInputs, {
+  Props
+} from '../../../../../components/admin/lessons/AdminModuleInputs'
+import Breadcrumbs from '../../../../../components/Breadcrumbs'
+import styles from '../../../../../scss/modules.module.scss'
+import navStyles from '../../../../../scss/adminLessonNav.module.scss'
+import { AdminLayout } from '../../../../../components/admin/AdminLayout'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+
+const MAIN_PATH = '/admin/lessons'
+
+const MODULES = gql`
+  query {
+    modules {
+      id
+      name
+      content
+      lesson {
+        id
+      }
+      order
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+  order: number
+}
+type Modules = Module[]
+
+type ReqParam = string | string[] | undefined
+
+type ContentProps = {
+  pageName: ReqParam
```

Comment:
Looks like the only place `ReqParam` used is here. Can we make pageName an optional field? That way you dont have to explicitly define ReqParam as having possibility of undefined yourself